### PR TITLE
Adds pull-request standards to the contributing guidelines

### PR DESCRIPTION
**Pull request by PowerfulBacon**
## About The Pull Request

Updates the contributing guidelines to add in pull-request standards.
Most of the things mentioned in this section are things that we should already be following.
The main things should be:
 - Maintainers may close PRs that do not follow the guidelines freely. They aren't required to, but have the choice if the author refuses to update the PR.
 - The testing section shouldn't be ignored for almost all PRs. Again, up to maintainer discretion if they want to close PRs that do not complete this section of the pull-request template, but it should exist as an option to encourage people to actually complete it.
 - Balance changes must be marked as balance and shouldn't hide the extent of their changes. I have noticed a few balance changes that only state the partial extent of the changes while hiding parts that may cause negative opinions of that pull request. (For example #6907 should mention that this affects all portals and not just hand-teleporter portals)

## Why It's Good For The Game

There are a large quantity of poorly tested PRs that are seemingly trivial, however result in large numbers of bugs that would have been easilly spotted if the changes were tested a single time. This gives maintainers more freedom to require authors to test their PRs before merging. Maintainers are here to ensure the quality of the code and shouldn't be the ones locating very obvious issues that would have been spotted with a basic compilation of the code.
All of the other things mentioned should already be happening, however this defines it clearly in the contributing guidelines.

## Testing Photographs and Procedure

<details>

You should complete the pull-request template in its entirity.

Any pull-request that does not adequately complete the pull-request template will not be merged and may be closed by maintainers (Updating the PR will allow for the pull-request to be reopened).
 - Balance changes must indicate that they affect the balance of the game.
 - Changes must be documented in their entirity, and the full extend of their effects mentioned. (For example, if you change it so all mobs are half speed, don't label the PR as 'monkeys now move twice as slow'). Failing to document the full extent of the changes may be seen as malicious and could result in a repo-ban in a worst case scenario.
 - The section labeled 'about this pull request' should state the changes that the pull-request makes.
 - The section labeled "why it's good for the game" should include the reasons behind the changes and how they will be good for the game.
 - The testing section should contain screenshots, videos or a description showing that the PR works as specified. Pull-requests that ignore this section or are not tested may be closed by maintainers. This applies to small PRs that may seem trivial.
 - The changelog should include a short summary of the changes made, as well as any other authors that contributed to the pull-request (Include their names after the :cl: tag along with your own)

</details>

## Changelog
:cl:
code: Updates the contributing guidelines to include a pull-request standards section.
/:cl:

### CODE REVIEW

File: .github/CONTRIBUTING.md
Comment (MNarath1) [inline .github/CONTRIBUTING.md]: This sounds a bit harsh i am not saying you should not document your changes but saying you can get repo banned just because you didn't properly write down a small change is a bit much also this doesn't really specifiy where you should document those changes after you made the PR(i personally like to put them in commit messages and if its major enough mention them in the about section) neither does it say how well you actually should document them it just provides a bad example no good example.

File: .github/CONTRIBUTING.md
Comment (MNarath1) [inline .github/CONTRIBUTING.md]: Now ok we all know this section has been ignored almost entirely since it got introduced(mainly cause its a hassle after hours of work writing down an essay of all possible stuff you did to test it) trough as long as a completly text documentation is (in most cases at least) sufficient i guess its fine.

File: .github/CONTRIBUTING.md
Comment (MNarath1) [reply .github/CONTRIBUTING.md]: Thinking about it i may have interpreted this a bit to harsh myself but mentioning where to document and what would count as a good example at least sounds like a good idea

**Review by Archanial — DISMISSED** [review]
Add something about needs wiki changes tag and making wiki changes at all.

File: .github/CONTRIBUTING.md
Comment (Archanial) [reply .github/CONTRIBUTING.md]: For it to be respected template needs to be improved.
It's currently not really readable and ignored by most.

File: .github/CONTRIBUTING.md
Comment (Archanial) [reply .github/CONTRIBUTING.md]: Something like "repeated violation" would be more appropriate here.

**Comment by Archanial** [comment]
Merge after announcing the changed in channel.

File: .github/CONTRIBUTING.md
Comment (PowerfulBacon) [reply .github/CONTRIBUTING.md]: It does say 'could' and that its a worse case scenario rather than you will be banned for this, but I can update it.

Comment (PowerfulBacon) [inline .github/CONTRIBUTING.md]:
```suggestion
 - Changes must be documented in their entirity, and the full extend of their effects mentioned. (For example, if you change it so all mobs are half speed, don't label the PR as 'monkeys now move twice as slow'). Failing to document the full extent of the changes may result in a repo-ban if the intent of hiding changes is seen as malicious.
```

File: .github/CONTRIBUTING.md
Comment (PowerfulBacon) [reply .github/CONTRIBUTING.md]: You just have to put a screenshot in it showing that you have opened it and done at least a basic test on it. An essay isn't required

**Comment by PowerfulBacon** [comment]
This should go through crossed

File: .github/CONTRIBUTING.md
Comment (Crossedfall) [inline .github/CONTRIBUTING.md]:
```suggestion
You should complete the pull-request template in its entirety.
```

File: .github/CONTRIBUTING.md
Comment (Crossedfall) [inline .github/CONTRIBUTING.md]:
```suggestion
Any pull-request that does not adequately complete the provided template may be closed or marked 'do not merge' by maintainers.
```

File: .github/CONTRIBUTING.md
Comment (Crossedfall) [inline .github/CONTRIBUTING.md]:
```suggestion
 - Changes must be documented in their entirety including the extent of their effects. (For example, if you change it so all mobs are half speed, don't label the PR as 'monkeys now move twice as slow'). Failing to document the full extent of the changes may result in a repo-ban if the intent of hiding changes is seen as malicious.
```

File: .github/CONTRIBUTING.md
Comment (Crossedfall) [reply .github/CONTRIBUTING.md:108]: > (mainly cause its a hassle after hours of work writing down an essay of all possible stuff you did to test it)

A basic set of reproduceable testing procedures is really all you need. Adding a screenshot or a video demonstrating that the thing works is equally acceptable and requires fairly minimal effort.

File: .github/CONTRIBUTING.md
Comment (Crossedfall) [inline .github/CONTRIBUTING.md]:
```suggestion
 - The testing section should contain screenshots, videos, and/or reproducible testing procedures showing that the PR works as specified. Pull-requests that ignore this section, or are not tested, may be closed by maintainers. This applies to small PRs that may seem trivial.
```

File: .github/CONTRIBUTING.md
Comment (Crossedfall) [inline .github/CONTRIBUTING.md]:
```suggestion
 - The changelog should include a short summary of the changes made; including any other authors that contributed to the pull-request (Include their names after the :cl: tag along with your own)
```

**Review by Crossedfall — CHANGES_REQUESTED** [review]
Quick pass

File: .github/CONTRIBUTING.md
Comment (Crossedfall) [reply .github/CONTRIBUTING.md]: > (mainly cause its a hassle after hours of work writing down an essay of all possible stuff you did to test it)

A basic set of reproduceable testing procedures is really all you need. Adding a screenshot or a video demonstrating that the thing works is equally acceptable and requires fairly minimal effort.

**Comment by PowerfulBacon** [comment]
Applied crossed review messages, added in a wiki section and updated the contents of the file.

File: .github/CONTRIBUTING.md
Comment (PowerfulBacon) [inline .github/CONTRIBUTING.md]:
```suggestion
If a pull-request requires updates to the wiki, these changes should be made on your user account page (For example: https://wiki.beestation13.com/view/User:PowerfulBacon), so that the original page can be updated on merge.
```

File: .github/CONTRIBUTING.md
Comment (MNarath1) [reply .github/CONTRIBUTING.md]: In that case yeah sounds fine

**Review by Crossedfall — DISMISSED** [review]

**Comment by Crossedfall** [comment]
Good by me. Other maintainers should sign off acknowledging that they read it.

**Review by PestoVerde322 — DISMISSED** [review]

**Review by ike709 — DISMISSED** [review]

File: .github/CONTRIBUTING.md
Comment (Archanial) [inline .github/CONTRIBUTING.md:112]: "or has needs wiki updates label"

**Review by Archanial — COMMENTED** [review]
One comment rest lgtm

File: .github/CONTRIBUTING.md
Comment (PowerfulBacon) [reply .github/CONTRIBUTING.md:112]: If a pull request has need wiki updates label then it can be assumed the pull-request requires updates to the wiki

Comment (Archanial) [reply .github/CONTRIBUTING.md:112]: Nothing wrong with explicitly stating this in contributing

Comment (Crossedfall) [reply .github/CONTRIBUTING.md:112]: I don't think it's needed.

Comment (PowerfulBacon) [reply .github/CONTRIBUTING.md:112]: I also don't think its necessary

File: .github/CONTRIBUTING.md
Comment (KubeRoot) [inline .github/CONTRIBUTING.md:106]: I feel like this is phrased too strongly. There are some things that are difficult to test due to needing complex setup, or infeasible to test fully due to the scope of changes. There are also cases where the person interested in contributing doesn't know how to test things properly (yet).

I think the wording should be eased off a little, and that we should instead require people to specify what they didn't test in those cases.

File: .github/CONTRIBUTING.md
Comment (KubeRoot) [inline .github/CONTRIBUTING.md]: The wording here is a bit awkward. I'd maybe suggest something like this

```suggestion
 - If you make any changes that affect the balance of the game, they must be mentioned as balance changes. Note that this also applies if you're fixing something, even if the result is what was originally intended.
```

File: .github/CONTRIBUTING.md
Comment (KubeRoot) [inline .github/CONTRIBUTING.md]: I feel like this point is too terse/unverbose, it might as well describe the changelog. I'd suggest either getting rid of it, or something more verbose like

```suggestion
 - The section labeled 'about this pull request' should describe the pull request in detail. This includes the change being made from the player's perspective, any important details about how it was implemented, as well as background on the pull request, such as the source of what might be ported or issues that are being fixed. Note that it should not include images and videos, there is a dedicated section for those.
```

File: .github/CONTRIBUTING.md
Comment (KubeRoot) [inline .github/CONTRIBUTING.md]: I would suggest a different wording here:

```suggestion
 - The changelog should include a short summary of the changes made. If your pull request includes things made by other people, you should list everybody who contributed, including yourself, after the :cl: tag.
```

**Review by KubeRoot — COMMENTED** [review]
In addition to my review comments, I would suggest a thing or two should be said about draft PRs. A lot of people don't know how to use them correctly, especially how to convert their own PR into a draft.

File: .github/CONTRIBUTING.md
Comment (Crossedfall) [reply .github/CONTRIBUTING.md:106]: This bit isn't about testing. It's about documenting changes and being honest about those changes.

File: .github/CONTRIBUTING.md
Comment (Crossedfall) [reply .github/CONTRIBUTING.md]: > Note that this also applies if you're fixing something, even if the result is what was originally intended.

This is poorly phrased. It implies that all fixes should be categorized as a balance change. I would drop that entire ending sentence. The alterations to the first sentence are good.

File: .github/CONTRIBUTING.md
Comment (PowerfulBacon) [reply .github/CONTRIBUTING.md]: About this pull request should allow for images and videos in it if the images and videos show the changes, I think thats too strong.

Imo the 'about this pull request' section should strictly be the what, the changes how they are in an objective manner, with the reason for implementing them going in the 'Why its good for the game section'.

The main reason for this bit is because all the other sections have a quick description too, about this pull request is pretty self-explanatory.

File: .github/CONTRIBUTING.md
Comment (PowerfulBacon) [inline .github/CONTRIBUTING.md]:
```suggestion
 - Any changes that may affect game balance should be documented as a balance change. Bug fixes that affect balance should be marked as a balance change if the bug could reasonably be seen as an intentional feature by somebody that has no knowledge of the code.
```

File: .github/CONTRIBUTING.md
Comment (KubeRoot) [reply .github/CONTRIBUTING.md:106]: Whoops, I misclicked. I started writing this earlier, but github dropped my unfinished review, and I didn't double-check when making this comment.

The line this was meant to be commented on is:
```
- The testing section should contain screenshots, videos, and/or reproducible testing procedures showing that the PR works as specified. Pull-requests that ignore this section, or are not tested, may be closed by maintainers. This applies to small PRs that may seem trivial.
```

File: .github/CONTRIBUTING.md
Comment (KubeRoot) [reply .github/CONTRIBUTING.md]: It might be poorly phrased, but I feel like we've had cases where significant balance changes were PR'd under the banner of fixes, because the original code's intent seemed to be different than what it actually did. What I mean is, even if the balance changes are the result of a fix, the rule should apply in the same way, treating it as a balance change based on what the difference is.

Comment (Crossedfall) [reply .github/CONTRIBUTING.md]: I think it's fine. Maintainers can edit PR bodies, so in those rare cases we can correct the CL as needed. I don't think it's worth going overly verbose and making it too confusion.

Comment (PowerfulBacon) [reply .github/CONTRIBUTING.md]: I proposed some other changes, but they went to the bottom of the comments since I didn't do it in this thread.

File: .github/CONTRIBUTING.md
Comment (KubeRoot) [reply .github/CONTRIBUTING.md]: > About this pull request should allow for images and videos in it if the images and videos show the changes, I think thats too strong.

This is a matter of personal opinion, but I was supportive of adding the `Screenshots&Videos` collapsible specifically because those tend to take up a lot of space, making browsing PRs painful. That is, it was my understanding that the collapsible is where screenshots and videos should go in general, and I had the impression we were in agreement about it.

> Imo the 'about this pull request' section should strictly be the what, the changes how they are in an objective manner, with the reason for implementing them going in the 'Why its good for the game section'.

I mostly agree, and I think the suggested changes do not contradict anything you said in this reply. Might be a case of bad wording, but for clarity, what I suggested it should be is:
- The actual changes, based on how they affect the users, in detail.
- Important implementation details. In cases of refactors/new systems/maintenance PRs, this would be the bulk of it, since they won't affect users much. In other PRs, this would only be needed when big/risky changes are being made, or implementation is changed in a way that's important to future coders.
- The background on the PR. It's not meant to be the reason for why it's good for the game, but rather elaboration on how the PR came to be. This is meant to be mainly PRs from other codebases being ported, Issues that are being fixed, previous PRs from this codebase that are being addressed in some ways. That said, if the author wants to cite other background, such as previous discussion about the matter, I feel like this is the good place to do it.

I also want to note that I see the "Why it's good for the game" as being a more strict section - to me, it shouldn't have vague elaboration, but specific reasoning for why it's an improvement. Because of this, anything like points to discuss further that are up to debate would be something to put in "About this pull request".

File: .github/CONTRIBUTING.md
Comment (Crossedfall) [reply .github/CONTRIBUTING.md:106]: See: https://github.com/BeeStation/BeeStation-Hornet/pull/6980#discussion_r885039673

File: .github/CONTRIBUTING.md
Comment (PowerfulBacon) [reply .github/CONTRIBUTING.md]: @KubeRoot @Crossedfall

File: .github/CONTRIBUTING.md
Comment (PowerfulBacon) [reply .github/CONTRIBUTING.md]:
```suggestion
 - The section labeled 'about this pull request' should describe the pull request's changes in detail. This includes the changes being made, any important details about how it was implemented, the issues it closes, and links to any other pull requests if code is being ported from another codebase.
```

File: .github/CONTRIBUTING.md
Comment (Crossedfall) [reply .github/CONTRIBUTING.md]:
```suggestion
 - Any changes that may affect game balance should be documented as a balance change. This also applies to bug fixes which directly alter the game's balance.
```

I think it'd be too difficult to determine what a layman would and wouldn't know is a balance change in this context. Probably better to just be broad.

File: .github/CONTRIBUTING.md
Comment (Crossedfall) [reply .github/CONTRIBUTING.md]: We're not animals- use the Oxford comma.

Comment (PowerfulBacon) [reply .github/CONTRIBUTING.md]: what is an oxford comma?

Comment (PowerfulBacon) [reply .github/CONTRIBUTING.md]: ew that is just wrong

Comment (Crossedfall) [reply .github/CONTRIBUTING.md]: You're objectively wrong. Add the comma, it's far more readable with it.

Comment (PowerfulBacon) [reply .github/CONTRIBUTING.md]: Fine

Comment (PowerfulBacon) [reply .github/CONTRIBUTING.md]: That good?

**Review by Crossedfall — APPROVED** [review]

**Review by qwertyquerty — APPROVED** [review]

**Pull request merged**
